File: backend/src/modeles/enrolement.py
# -*- coding: utf-8 -*-
"""
Modèle Enrôlement citoyen — Géré par les agents terrain.
"""
import uuid
from datetime import datetime

from sqlalchemy import Column, String, Text, DateTime, ForeignKey, Boolean
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship

from src.base_donnees.base import Base


class Enrolement(Base):
    __tablename__ = "enrolements"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    agent_id = Column(UUID(as_uuid=True), ForeignKey("utilisateurs.id"), nullable=False, index=True)
    citoyen_nom = Column(String(255), nullable=False)
    citoyen_prenom = Column(String(255), nullable=False)
    citoyen_digiid = Column(String(50), nullable=True, index=True)
    citoyen_telephone = Column(String(50), nullable=True)
    citoyen_email = Column(String(255), nullable=True)
    statut = Column(String(20), default="en_attente")  # en_attente, valide, rejete
    notes = Column(Text, nullable=True)
    scan_cni = Column(Boolean, default=False)
    capture_biometrique = Column(Boolean, default=False)
    date_enrolement = Column(DateTime, default=datetime.utcnow, nullable=False)
    date_validation = Column(DateTime, nullable=True)

    agent = relationship("Utilisateur", backref="enrolements")

File: frontend/src/services/enrolement.ts
/**
 * Service API pour l'enrôlement citoyen (agent terrain).
 */
import { clientAPI } from "./client_api";

export interface Enrolement {
  id: string;
  agent_id: string;
  citoyen_nom: string;
  citoyen_prenom: string;
  citoyen_digiid: string | null;
  citoyen_telephone: string | null;
  citoyen_email: string | null;
  statut: string;
  notes: string | null;
  scan_cni: boolean;
  capture_biometrique: boolean;
  date_enrolement: string;
  date_validation: string | null;
}

export async function listerEnrolements(statut?: string): Promise<Enrolement[]> {
    const params = statut && statut !== "tous" ? `?statut=${statut}` : "";
    return clientAPI.get<Enrolement[]>(`/api/v1/enrolement/liste${params}`, {
      authentifie: true,
    });
  }
  
  export async function creerEnrolement(data: {
    citoyen_nom: string;
    citoyen_prenom: string;
    citoyen_telephone: string;
    citoyen_email?: string;
    notes?: string;
  }): Promise<Enrolement> {
    return clientAPI.post<Enrolement>("/api/v1/enrolement/creer", data, {
      authentifie: true,
    });
  }
  export async function modifierEnrolement(
    id: string,
    data: { statut?: string; scan_cni?: boolean; capture_biometrique?: boolean; notes?: string },
  ): Promise<Enrolement> {
    return clientAPI.patch<Enrolement>(`/api/v1/enrolement/${id}`, data, {
      authentifie: true,
    });
}
